src/commands/lambda/flare.ts

```diff
@@ -86,61 +94,132 @@ export class LambdaFlareCommand extends Command {
+    // Get and print Lambda function configuration
+    this.context.stdout.write('\n🔍 Fetching Lambda function configuration...\n')
     const lambdaClientConfig: LambdaClientConfig = {
       region,
       credentials: this.credentials,
     }
     const lambdaClient = new LambdaClient(lambdaClientConfig)
     let config
     try {
-      config = await getLambdaFunctionConfig(lambdaClient, this.functionName ?? '')
+      config = await getLambdaFunctionConfig(lambdaClient, this.functionName)
     } catch (err) {
-      this.context.stderr.write(
-        commonRenderer.renderError(`Unable to get Lambda function configuration: ${err.message}`)
-      )
+      if (err instanceof Error) {
+        this.context.stderr.write(
+          commonRenderer.renderError(`Unable to get Lambda function configuration: ${err.message}`)
+        )
+      }
 
       return 1
     }
+    const configStr = util.inspect(config, false, undefined, true)
+    this.context.stdout.write(`\n${configStr}\n`)
 
-    // Print config
-    const configStrColored = util.inspect(config, false, undefined, true)
-    this.context.stdout.write(`\n${configStrColored}\n`)
-    if (this.isDryRun) {
-      this.context.stdout.write('\n🚫 The configuration was not sent as it was executed in dry run mode.\n')
+    // Get CloudWatch logs
+    let logs: Map<string, OutputLogEvent[]> = new Map()
+    if (this.withLogs) {
+      this.context.stdout.write('\n☁️ Getting CloudWatch logs...\n')
+      try {
+        logs = await getAllLogs(region!, this.functionName)
+      } catch (err) {
+        if (err instanceof Error) {
+          this.context.stderr.write(commonRenderer.renderError(err.message))
+        }
 
-      return 0
+        return 1
+      }
     }
 
-    // Send data to Datadog
-    const folderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
-    const filePath = path.join(folderPath, FUNCTION_CONFIG_FILE_NAME)
-    const zipPath = path.join(folderPath, ZIP_FILE_NAME)
-    const configStrUncolored = JSON.stringify(config, undefined, 2)
     try {
-      await writeFile(folderPath, filePath, configStrUncolored, this.context)
-      await zipContents(filePath, zipPath)
+      // CloudWatch messages
+      if (this.withLogs) {
+        if (logs.size === 0) {
+          this.context.stdout.write(
+            commonRenderer.renderSoftWarning(
+              'No CloudWatch log streams were found. Logs will not be retrieved or sent.'
+            )
+          )
+        } else {
+          this.context.stdout.write('\n✅ Found log streams:\n')
+          for (const [logStreamName, logEvents] of logs) {
+            const warningMessage =
+              logEvents && logEvents.length > 0
+                ? '\n'
+                : ' - ' + commonRenderer.renderSoftWarning('No log events found in this stream')
+            this.context.stdout.write(`• ${logStreamName}${warningMessage}`)
+          }
+        }
+      }
```

Comment:
This part is not very readable, but I think it might be necessary. Do you have any suggestions on how to improve it?

Reply:
Following this suggestion, could you find a way to remove the if-else block?
```suggestion
      if (this.withLogs) {
        if (logs.size === 0) {
          this.context.stdout.write(
            commonRenderer.renderSoftWarning(
              'No CloudWatch log streams were found. Logs will not be retrieved or sent.'
            )
          )
        } else {
          this.context.stdout.write('\n✅ Found log streams:\n')
          for (const [logStreamName, logEvents] of logs) {
            let warningMessage = '\n'
            if (logEvents.length === 0) {
                warningMessage = ' - ' + commonRenderer.renderSoftWarning('No log events found in this stream')
            }
            this.context.stdout.write(`• ${logStreamName}${warningMessage}`)
          }
        }
      }
```

Reply:
I have moved the for loop outside of the if/else block. WDYT?
```
if (this.withLogs) {
        if (logs.size === 0) {
          this.context.stdout.write(
            commonRenderer.renderSoftWarning(
              'No CloudWatch log streams were found. Logs will not be retrieved or sent.'
            )
          )
        } else {
          this.context.stdout.write('\n✅ Found log streams:\n')
        }

        for (const [logStreamName, logEvents] of logs) {
          let warningMessage = '\n'
          if (logEvents.length === 0) {
            warningMessage = ' - ' + commonRenderer.renderSoftWarning('No log events found in this stream')
          }
          this.context.stdout.write(`• ${logStreamName}${warningMessage}`)
        }
      }
```